crates/uv-cli/src/lib.rs

```diff
@@ -2635,6 +2635,13 @@ pub struct BuildArgs {
     #[arg(long, conflicts_with = "list")]
     pub force_pep517: bool,
 
+    #[arg(long, overrides_with("no_gitignore"), hide = true)]
+    pub gitignore: bool,
+
+    /// Do not create a `.gitignore` file in the output directory
+    #[arg(long, overrides_with("gitignore"))]
+    pub no_gitignore: bool,
```

Comment:
My intuition when I saw this flag is that it would toggle whether or not gitignore is respected during the build. I think this could be confusing enough (and need for this option seems rare enough) that I would call this `--no-create-gitignore` / `--create-gitignore`.